tests/src/test/scala/jms4s/jms/JmsSpec.scala

```diff
@@ -41,30 +41,23 @@ class JmsSpec extends AsyncFreeSpec with AsyncIOSpec with Matchers with Jms4sBas
       }
     }
     "publish and then receive with a delay" in {
-      def receiveAfter(received: Ref[IO, Set[String]], duration: FiniteDuration) =
-        for {
-          _        <- IO.sleep(duration / 2)
-          tooEarly <- received.get
-          _ <- if (tooEarly.nonEmpty)
-                IO.raiseError(new RuntimeException("Delay has not been respected!"))
-              else
-                IO.unit
-          _      <- IO.sleep(duration / 2)
-          gotcha <- received.get
-        } yield gotcha
-
       queueRes.use {
-        case (queueConsumer, queueProducer, msg) =>
+        case (consumer, producer, msg) =>
           for {
-            _        <- queueProducer.setDeliveryDelay(delay)
-            _        <- queueProducer.send(msg)
-            received <- Ref.of[IO, Set[String]](Set())
-            consumerFiber <- (for {
-                              body <- receiveBodyAsTextOrFail(queueConsumer)
-                              _    <- received.update(_ + body)
-                            } yield ()).start
-            gotcha <- receiveAfter(received, delay).guarantee(consumerFiber.cancel)
-          } yield gotcha.shouldBe(Set("body"))
+            _                 <- producer.setDeliveryDelay(delay)
+            producerTimestamp <- IO(System.currentTimeMillis())
+            _                 <- producer.send(msg)
+            msg               <- consumer.receiveJmsMessage
+            tm                <- msg.asJmsTextMessage
+            body              <- tm.getText
+            jmsDeliveryTime   <- tm.getJMSDeliveryTime
+            producerDelay     <- IO(jmsDeliveryTime - producerTimestamp)
+          } yield (body, producerDelay)
+            .shouldBe(
+              new ResultOfGreaterThanOrEqualToComparison(
+                (body, delay.toMillis)
```

Comment:
We are checking `body` with.. `body` itself?

Reply:
One last thing: I'd rewrite the assertion like `assert(producerDelay >= delay.toMillis && body == expectedBody)`

Reply:
Thanks!